### PR TITLE
[IS 5.2.0] Multi Tenant Support and Handler wise Implementation for Identity Management

### PR DESCRIPTION
**Pull request by IndunilRathnayake**

### CODE REVIEW

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt.admin.ui/pom.xml [hunk 1]
```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+~ Copyright (c) 2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+~
```
Comment (isharak) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt.admin.ui/pom.xml]: 2010 should be changed

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt.admin.ui/pom.xml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                    <instructions>
+                        <Bundle-SymbolicName>
+                            ${project.artifactId}
+                        </Bundle-SymbolicName>
+                        <Export-Package>
+                            org.wso2.carbon.identity.mgt.admin.ui.*,
+                        </Export-Package>
+                        <Import-Package>
+                        </Import-Package>
+                        <DynamicImport-Package>*</DynamicImport-Package>
```
Comment (isharak) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt.admin.ui/pom.xml]: Remove dynamic imports and add relevant import packages

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt.admin.ui/src/main/java/org/wso2/carbon/identity/mgt/admin/ui/client/TenantIdentityMgtClient.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    protected UserIdentityManagementAdminServiceStub stub = null;
+
+    protected static Log log = LogFactory.getLog(TenantIdentityMgtClient.class);
+
+    public TenantIdentityMgtClient(String cookie, String backendServerURL,
+                                   ConfigurationContext configContext)
+            throws Exception {
+        try {
+            stub = new UserIdentityManagementAdminServiceStub(configContext, backendServerURL + "UserIdentityManagementAdminService");
```
Comment (isharak) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt.admin.ui/src/main/java/org/wso2/carbon/identity/mgt/admin/ui/client/TenantIdentityMgtClient.java]: define a constant for this

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt.admin.ui/src/main/resources/META-INF/component.xml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+<!--
+~ Copyright (c) 2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+~
```
Comment (isharak) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt.admin.ui/src/main/resources/META-INF/component.xml]: Year should be changed 2010

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt.admin.ui/src/main/resources/web/identity-tenantmgt/add_config_ajaxprocessor.jsp [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            add("UserAccount.Verification.Enable");
+            add("Temporary.Password.Enable");
+            add("Temporary.Password.Default.Value");
+            add("Authentication.Policy.Account.Lock.On.Creation");
+        }};
+
+
+        HashMap<String, String> configMap = new HashMap<String, String>();
+
+        for (int i = 0; i < configNames.size(); i++) {
```
Comment (isharak) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt.admin.ui/src/main/resources/web/identity-tenantmgt/add_config_ajaxprocessor.jsp]: You can use enhanced for loop here

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt/pom.xml [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+                            org.wso2.carbon.user.api; version="${carbon.user.api.imp.pkg.version.range}",
+                            org.wso2.carbon.registry.core.*;version="${carbon.kernel.registry.imp.pkg.version}",
+
+                            org.wso2.carbon.identity.core.*; version="${carbon.identity.package.import.version.range}",
+                            org.wso2.carbon.identity.base; version="${carbon.identity.package.import.version.range}",
+                            org.wso2.carbon.captcha.mgt.*;
+                            version="${carbon.captcha.mgt.imp.pkg.version.range}",
+                        </Import-Package>
+                        <DynamicImport-Package>*</DynamicImport-Package>
```
Comment (isharak) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt/pom.xml]: Remove dynamic imports

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConfig.java [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+    public Properties getConfiguration(int tenantId) throws IdentityMgtException{
+
+        Properties properties = new Properties();
+        Map<String, String> configMap = null;
+        try {
+            configMap = identityMgtService.getConfiguration(tenantId);
+        } catch (IdentityMgtException ex) {
+            String msg = "Error when retrieving configurations";
+            log.error(msg);
```
Comment (isharak) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConfig.java]: Don't log and throw

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConfig.java [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+            throw new IdentityMgtException(msg,ex);
+        }
+        Iterator<Map.Entry<String, String>> iterator = configMap.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<String, String> pair = iterator.next();
+            properties.setProperty(pair.getKey(), pair.getValue());
+            System.out.println(pair.getKey() + " = " + pair.getValue());
```
Comment (isharak) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConfig.java]: System.out.println ??

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt/pom.xml [hunk 9]
```diff
@@ -0,0 +1,134 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+~ Copyright (c) 2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (cdwijayarathna) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt/pom.xml]: 2015

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConfigGlobal.java [hunk 10]
```diff
@@ -0,0 +1,189 @@
+package org.wso2.carbon.identity.mgt;
```
Comment (cdwijayarathna) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConfigGlobal.java]: add licence

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConstants.java [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.mgt;
+
+/**
+ * Identity management related constants
+ */
+public class IdentityMgtConstants {
```
Comment (cdwijayarathna) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConstants.java]: add private constructors to this and inner classes

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/policy/PolicyEnforcer.java [hunk 12]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (cdwijayarathna) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/policy/PolicyEnforcer.java]: 2015?

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/policy/PolicyRegistry.java [hunk 13]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (cdwijayarathna) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/policy/PolicyRegistry.java]: 2015?

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/service/IdentityMgtServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+import java.io.IOException;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.List;
+
+public class IdentityMgtServiceImpl implements IdentityMgtService {
+
+    Log log = LogFactory.getLog(IdentityMgtServiceImpl.class);
```
Comment (cdwijayarathna) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/service/IdentityMgtServiceImpl.java]: private?

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/UserRecoveryDataStore.java [hunk 15]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (cdwijayarathna) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/UserRecoveryDataStore.java]: 2015

File: components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/resources/META-INF/services.xml [hunk 16]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+~ Copyright (c) 2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (cdwijayarathna) [inline components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/resources/META-INF/services.xml]: 2015

**Pull request closed without merging**